### PR TITLE
even more minesweeping fixes

### PR DESCRIPTION
**Pull request by Jujumatic**
hi theer it seems my win condition actually got deleted... hm
yeah ok

also added the click sound when flagging squares and clicking the button to start flagging shit

## Changelog
:cl:
tweak: Added the board click sound when flagging minesweeper squares
fix: You can now actually win minesweeper
/:cl:

### CODE REVIEW

**Comment by yoyobatty** [comment]
Your CL isn't exactly descriptive, people will just see "Win condition now works properly... again" in the ingame CL and wonder what that means.

**Comment by Jujumatic** [comment]
It was a long day yoyo

File: hippiestation/code/game/machinery/computer/arcade.dm [hunk 1]
```diff
@@ -183,14 +184,14 @@
 		mine_sound = TRUE
 
 	area = (rows-1)*(columns-1)
-	safe_squares_revealed = 0
-	win_condition = area-mine_placed
 
 	if(reset_board)
 		mine_placed = 0
 		var/reset_everything = TRUE
```
Comment (yoyobatty) [inline hippiestation/code/game/machinery/computer/arcade.dm:190]: What is the point of this var? I don't see it getting used outside this if statement

**Review by yoyobatty — APPROVED** [review]
It's not like you added shit code, but the previous code isn't so good, so I'll let it slide.

File: hippiestation/code/game/machinery/computer/arcade.dm (hunk 1 above)
Comment (Jujumatic) [reply hippiestation/code/game/machinery/computer/arcade.dm:190]: every square that has a value above 10 aka revealed and not a mine will add to the counter of how many safe squares are revealed. if the win condition aka all squares minus the placed mines is greater than or equal to the number of safe squares revealed, you win! i just had to move this part somewhere else as it was getting its value set at the wrong time

**Pull request merged**